torch/distributions/exp_family.py

```diff
@@ -0,0 +1,80 @@
+import torch
+from torch.distributions.distribution import Distribution
+from torch.autograd import Variable
+
+
+class ExponentialFamily(Distribution):
+    r"""
+    ExponentialFamily is the abstract base class for probability distributions belonging to an
+    exponential family, whose probability mass/density function has the form is defined below
+
+    ..math::
+
+        p_{F}(x; \theta) = \exp(\langle t(x), \theta\rangle) - F(\theta) + k(x))
+
+    where :math:`\theta` denotes the natural parameters, :math:`t(x)` denotes the sufficient statistic,
+    :math:`F(\theta)` is the log normalizer function for a given family and :math:`k(x)` is the carrier
+    measure.
+
+    Note:
+        This class is an intermediary between the `Distribution` class and distributions which belong
+        to an exponential family mainly to check the correctness of the `.entropy()` and analytic KL
+        divergence methods. We use this class to compute the entropy and KL divergence using the AD frame-
+        work and Bregman divergences (courtesy of: Frank Nielsen and Richard Nock, Entropies and
+        Cross-entropies of Exponential Families).
+    """
+
+    @property
+    def natural_params(self):
+        """
+        Abstract method for natural parameters. Returns a tuple of Variables based
+        on the distribution
+        """
+        raise NotImplementedError
+
+    def log_normalizer(self):
+        """
+        Abstract method for log normalizer function. Returns a callable lambda to compute
+        the log normalizer based on the distribution and input
+        """
+        raise NotImplementedError
+
+    @property
+    def mean_carrier_measure(self):
```

Comment:
Is it the case that `mean_carrier_measure` never depends on params? If it never depends on params, then this implementation works fine. However if it ever does depend on params, then you should change this to a function with the same signature as `log_normalizer()`.

Reply:
Expected Carrier measure is independent of natural params.

Reply:
At least for the distributions we have now, `mean_carrier_measure` is independent of `natural_params`.